shuup/core/basket/objects.py

```diff
@@ -208,6 +208,38 @@ def billing_address(self, value):
             self._load()['billing_address_id'] = value.id
 
     @property
+    def shipping_method(self):
+        if not self.shipping_method_id:
+            if hasattr(self, "_data") and self._load().get('shipping_method_id'):
+                self.shipping_method_id = self._load()['shipping_method_id']
+
+        if self.shipping_method_id:
+            return ShippingMethod.objects.get(pk=self.shipping_method_id)
+
+    @shipping_method.setter
+    def shipping_method(self, shipping_method):
+        self.shipping_method_id = (shipping_method.id if shipping_method else None)
+
+        if self.shipping_method_id and hasattr(self, "_data"):
+            self._load()['shipping_method_id'] = self.shipping_method_id
```

Comment:
yeah, this is also at the old code, but this logic do not accept to clearing the shipping method right? this also happens with addresses and payment method. maybe we should accept `None` here?